---
title: "microbiome report"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load required packages

```{r}
library(dada2)
library(dplyr)
library(ggplot2)
```
# Load sequences
```{r}
path <- "~/microbiome_data"
list.files(all.files = TRUE)
```
# read in file names

```{r cars}
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`,1)
```

## Inspect read quality
```{r}
plotQualityProfile(fnFs[1:2])
```
```{r}
plotQualityProfile(fnRs[1:2])
```

#filter and trim

```{r pressure, echo=FALSE}
filtFs <- file.path(path, "filtered", paste(sample.names, "microbiome_data"))
names(filtFs) <- sample.names
out <- filterAndTrim(fnFs, filtFs, truncLen=c(120), maxN=0, maxEE=c(2), tuncQ=2, rm.phix=TRUE, compress=TRUE, multithread=FALSE)

```

# learn error rates

```{r}
errF <- learnErrors(filtFs, multithread=FALSE)
```
```{r}
plotErrors(errF, nominalQ = TRUE)
```
```{r}
dadaFs <- dada(filtFs, err=errF, multithread = FALSE)
```
#sample inference
```{r}
dadaFs <- dada(filtFs, err=errF, multithread=FALSE)
```

#create sequence table
```{r}
seqtab <- makeSequenceTable(dadaFs)
dim(seqtab)
```
#inspect distribution of sequence lengths
```{r}
table(nchar(getSequences(seqtab)))
```
#remove chimera
```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=FALSE, verbose = TRUE)
```
```{r}
dim(seqtab.nochim)
```
#track reads through pipeline
```{r}
getN <- function(x) sum(getUniques(x))
tracks <- cbind(out), sapply(dadaFs, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "nonchim")
rownames(track) <- sample.names
head(track)
```
# save seqtab.nochim as an R file
```{r}
save(seqtab.nochim, file="RData/seqtab.nochim.RData")
```
# load seqtab.nochim to start here
```{r}
load("RData/seqtab.nochim.RData")
```
# assign taxonomy
```{r}
taxa <- assignTaxonomy(seqtab.nochim, "")
```
# save taxonomy as a file
```{r}
save(taxa, file = "RData/taxa.RData")
```
# Load taxa and seqtab.nochim
```{r}
load("RData/taxa.RData")
load("RData/seqtab.nochim.RData")
```
#import metadata
```{r}
metadata<-read.csv("metadata.tsv", header =TRUE, row.names = 1)
```
#create phyloseq object

```{r}
phyloseq <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), sample_data(metadata), tax_table(taxa))
phsyeq
```
#remove the sequence itself and replace with ASV
```{r}
dna <- Biostrings::DNAStringSet(taxa_names(physeq))
physeq
```
#remove the mitochondria and chloroplast matches and remove all non bacterial sequences
```{r}
physeq<- physique %>% subset_taxa(Family!= "Mitochondria" | is.na(Family) & Order! = "Chloroplast" | is.na(Order))
physeq
```
# remove all non bacterial sequences
```{r}
physeq<-rm_nonbac(physeq)
physeq
```
#save physeq objects to load later 
```{r}
save(physeq, file="RData/physeq.RData")
```
# load physeq objects to start here
```{r}
load("RData/physeq.RData")
```
# plot bar graph based on phylum
```{r}
plot_bar(physeq, fill="Phylum") + geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="fill")
```
# create a barplot of relative abundance
```{r}
physeq_relabund <- transform_sample_counts(physeq, function(x) x / sum(x))
```
# bar plot
```{r}
plot_bar(physeq_relabund, fill="Phylum") + geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="fill") + facet_wrap(~body.site, scales= "free")
```
# plot alpha diversity based on body site
```{r}
plot_richness(physeq, x="body.site", color="subject", measures=c("Observed", "Simpson", "Shannon"))
```
# plot alpha diversity based on reported antibiotic usage
```{r}
plot_richness(physeq, x="reported.antibiotic.usage", color="body.site", measures=c("Observed", "Simpson", "Shannon"))
```
# plot alpha diversity based on subject
```{r}
plot_richness(physeq, x="subject", color="body.site", measures=c("Observed", "Simpson", "Shannon"))
```
# test for normality
```{r}
alpha <- estimate_richness(physeq, measures=c("Observed", "Simpson", "Shannon"))
```
# shapiro-wilk
```{r}
observed <- shapiro.test(alpha$Observed)
shannon <- shapiro.test(alpha$Shannon)
simpson <- shapiro.test(alpha$Simpson)
```
# print
```{r}
print(observed)
print(shannon)
print(simpson)
```
# create data frames for statistical analyses
```{r}
samples <- sample_data(physeq)
if (class(samples) =="sample_data") {samples <- data.frame(sample_data(samples))}
alpha$sample <- rownames(alpha)
alpha <- merge(alpha, samples, by = "sample")
```
# perform statistics based on subject
```{r}
test_observed <- wilcox.test(Observed ~ subject, data=alpha)
test_simpson <- wilcox.test(Simpson ~ subject, data=alpha)
test_shannon <- t.test(Shannon ~ subject, data=alpha)
print(test_observed)
print(test_simpson)
print(test_shannon)
```
# perform statistics based on reported antibiotic usage
```{r}
test_observed <- wilcox.test(Observed ~ reported.antibiotic.usage, data=alpha)
test_simpson <- wilcox.test(Simpson ~ reported.antibiotic.usage, data=alpha)
test_shannon <- t.test(Shannon ~ reported.antibiotic.usage, data=alpha)
print(test_observed)
print(test_simpson)
print(test_shannon)
```
# test for body site
```{r}
kruskal.test(Simpson ~ body.site, data=alpha)
pairwise.wilcox.test(alpha$Simpson, alpha$body.site, p.adjust.method = "holm")
kruskal.test(Observed ~ body.site, data=alpha)
pairwise.wilcox.test(alpha$Observed, alpha$body.site, p.adjust.method = "holm")
shannonanova <- aov(Shannon ~ body.site, data=alpha)
summary(shannonanova)
TukeyHSD(shannonanova)
```
# remove taxa with relative abundance <0.005%
```{r}
minTotRelAbun = .00005
x = taxa_sums(physeq)
keepTaxa = (x/sum(x)) > minTotRelAbun
Phyloseqprune = prune_taxa(keepTaxa , physeq)
Physeqprune
```
# number oof shared asvs body site 
```{r}
bodysite=ps_venn(
Physeqprune,
"body.site" ,
fraction = .25,
weight = FALSE,
relative = TRUE,
Plot = TRUE
)
bodysite
```
# bray curtis calculation
```{r}
braysubject=plot_ordination(physeqprune, ordination, color="subject") +
theme_classic() + 
theme(strip.background = element_blank()) + stat_ellipse(aes(group=subject))
braysubject
```
# bray curtis body site plot
```{r}
braybodysite=plot_ordination(physeqprune, ordination, color = "body.site") + 
theme_classic() +
theme(strip.background = element_blank()) + stat_ellipse(aes(group=subject))
braysubject
```
# bray curtis subject statistics
```{r}
adonisZ(dist - sample_data(physeqprune)$subject)
```
# bray curtis site statistics
```{r}
adonisZ(dist - sample_data(physeqprune)$body.site)
Ps.disper<-betadisper(dist, sample_data(physeqprune)$body.site)
permutest(ps.disper, pair=TRUE)
```

